docker/Dockerfile

```diff
@@ -1,55 +1,81 @@
-  && chmod +x build_nceplibs.sh \
-  && cd .. \
-  && ./contrib/build_nceplibs.sh $PWD/nceplibs
-
-ENV bacio_ROOT /comsoftware/ccpp-scm/nceplibs
-ENV sp_ROOT /comsoftware/ccpp-scm/nceplibs
-ENV w3nco_ROOT /comsoftware/ccpp-scm/nceplibs
-
-# Create your own link from python -> python3 
-# This works without setting the system PATH env var
-# since /usr/local/bin is before /usr/bin in the search path.
-USER root
-RUN ln -s /usr/bin/python3 /usr/local/bin/python
+MAINTAINER Grant Firl <grantf@ucar.edu> or Michael Kavulich <kavulich@ucar.edu>
+
+#Compiler environment variables
+ENV CC  /usr/bin/gcc
+ENV FC  /usr/bin/gfortran
+ENV CXX /usr/bin/g++
+ENV F77 /usr/bin/gfortran
+ENV F90 /usr/bin/gfortran
+
+# Other necessary environment variables
+ENV LD_LIBRARY_PATH /usr/lib/
+
+# Set up unpriviledged user account, set up user home space and make sure user has permissions on all stuff in /comsoftware
+RUN groupadd comusers -g 9999 \
+ && useradd -u 9999 -g comusers -M -s /bin/bash -c "Unpriviledged user account" -d /home comuser \
+ && mkdir /comsoftware \
+ && chown -R comuser:comusers /home \
+ && chmod 6755 /home \
+ && chown -R comuser:comusers /comsoftware \
+ && chmod -R 6755 /comsoftware
+
+# Link version-specific aliases (python3 will be created later with virtual environment)
+RUN ln -s ~comuser/.venv/bin/python3 /usr/local/bin/python
+RUN ln -s /usr/bin/gfortran-12 /usr/bin/gfortran
+
+# all root steps completed above, now continue below as regular userID comuser
 USER comuser
+WORKDIR /home
+
+# Build NCEP libraries we need for SCM
+
+ENV NCEPLIBS_DIR /comsoftware/nceplibs
 
-# Invoke cmake on the source code to build
-RUN cd /comsoftware/ccpp-scm/scm \
+RUN mkdir -p $NCEPLIBS_DIR/src && cd $NCEPLIBS_DIR/src \
+  && git clone -b v2.4.1 --recursive https://github.com/NOAA-EMC/NCEPLIBS-bacio \
+  && mkdir NCEPLIBS-bacio/build && cd NCEPLIBS-bacio/build \
+  && cmake -DCMAKE_INSTALL_PREFIX=$NCEPLIBS_DIR .. \
+  && make VERBOSE=1 \
+  && make install 
+
+RUN cd $NCEPLIBS_DIR/src \
+  && git clone -b v2.3.3 --recursive https://github.com/NOAA-EMC/NCEPLIBS-sp \
+  && mkdir NCEPLIBS-sp/build && cd NCEPLIBS-sp/build \
+  && cmake -DCMAKE_INSTALL_PREFIX=$NCEPLIBS_DIR .. \
+  && make VERBOSE=1 \
+  && make install
+
+RUN cd $NCEPLIBS_DIR/src \
+  && git clone -b v2.11.0 --recursive https://github.com/NOAA-EMC/NCEPLIBS-w3emc \
+  && mkdir NCEPLIBS-w3emc/build && cd NCEPLIBS-w3emc/build \
+  && cmake -DCMAKE_INSTALL_PREFIX=$NCEPLIBS_DIR .. \
+  && make VERBOSE=1 \
+  && make install
+
+ENV bacio_ROOT /comsoftware/nceplibs
+ENV sp_ROOT /comsoftware/nceplibs
+ENV w3emc_ROOT /comsoftware/nceplibs
+
+# Obtain CCPP SCM source code and static data, build code
+RUN cd /comsoftware \
+  && git clone --recursive -b main https://github.com/NCAR/ccpp-scm \
+  && cd /comsoftware/ccpp-scm/ \
+  && ./contrib/get_all_static_data.sh \
+  && ./contrib/get_thompson_tables.sh \
```

Comment:
It's possible that we'll also need to add the ./get_aerosol_climo.sh script here too if we need to use GOCART climatology for anything in the Docker container. If the use case is just to show that it works, then I guess not, but if we need to run the HR3 suite that uses GOCART climatology, then we'll need to add it. Unfortunately, it's over 1 GB.

Reply:
I had thought about that...I think it's outside the scope of this PR, but one solution would be to "bind mount" a local directory to allow users to download this data separately outside the container (and not require downloading the data at build-time).